src/lib/Tab/Event/Subscriber/ConditionalTabSubscriber.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Tab\Event\Subscriber;
+
+use EzSystems\EzPlatformAdminUi\Tab\ConditionalTabInterface;
+use EzSystems\EzPlatformAdminUi\Tab\Event\TabEvents;
+use EzSystems\EzPlatformAdminUi\Tab\Event\TabGroupEvent;
+use EzSystems\EzPlatformAdminUi\Tab\OrderedTabInterface;
+use EzSystems\EzPlatformAdminUi\Tab\TabInterface;
+use EzSystems\EzPlatformAdminUi\UI\Service\TabService;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Reorders tabs according to their Order value (Tabs implementing OrderedTabInterface).
+ * Tabs without order specified are pushed to the end of the group.
+ *
+ * @see OrderedTabInterface
+ */
+class ConditionalTabSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var \EzSystems\EzPlatformAdminUi\UI\Service\TabService
+     */
+    private $tabService;
+
+    public function __construct(TabService $tabService)
+    {
+        $this->tabService = $tabService;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            TabEvents::TAB_GROUP_INITIALIZE => ['onTabGroupInitialize'],
+        ];
+    }
+
+    /**
+     * @param TabGroupEvent $tabGroupEvent
+     */
+    public function onTabGroupInitialize(TabGroupEvent $tabGroupEvent)
+    {
+        $tabGroup = $tabGroupEvent->getData();
+        $parameters = $tabGroupEvent->getParameters();
+        $tabGroup = $this->tabService->getTabGroup($tabGroup->getIdentifier());
+        $tabs = $this->processTabs($tabGroup->getTabs(), $parameters);
+
+        $tabGroup->setTabs($tabs);
+        $tabGroupEvent->setData($tabGroup);
+    }
+
+    /**
+     * @param TabInterface[] $tabs
+     * @param array $parameters
+     *
+     * @return array
+     */
+    private function processTabs($tabs, $parameters): array
+    {
+        foreach ($tabs as $tab) {
+            if ($tab instanceof ConditionalTabInterface && !$tab->evaluate($parameters)) {
+                unset($tabs[$tab->getIdentifier()]);
```

Comment:
As we have `TAB_GROUP_INITIALIZE` now, I would not go this way. I would reverse this logic as it make much more sense to add tabs if they match condition instead of removing tabs which don't.

I would remove this `processTabs` method and move foreach to `onTabGroupInitialize` with something like this:

```
foreach ($tabs as $tab) {
    if ($tab instanceof ConditionalTabInterface && $tab->evaluate($parameters)) {
        $tabGroup->addTab($tab);
    }
}
```